src/js/index.js

```diff
@@ -1,9 +1,19 @@
+var $ = function (query) {
+  var res = document.querySelectorAll(query);
+
+  if (res.length === 1) {
+    return res[0];
+  }
+
+  return res;
+};
```

Comment:
If we have n number of elements on a page where n >= 1, does this return either a NodeList in the event that n > 1 or a Node if n === 1? If the NodeList returned from querySelectorAll could be either 1 or greater than 1 does this return two different interfaces, a Node or a NodeList respectively? If I'm not mistaken, I think doing so would force us to check the length property on the returned value, and write a conditional that handles both situations if we do not definitively know that there is only one node in the DOM or many nodes.

Reply:
Yeah, it's a Node if n === 1 and a NodeList if n > 1. I use it as shorthand and find it extremely helpful, as you can often intuit if your selector matches one or multiple Nodes.

Reply:
It's near identical to the jQuery functionality.